polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -168,13 +180,44 @@ pub fn worker_entrypoint(
 					Arc::clone(&condvar),
 					WaitOutcome::TimedOut,
 				)?;
+
+				let fd = stream.as_raw_fd();
+				let crash_event = unsafe {
+					crash_handler::make_crash_event(move |context| {
+						ALLOC.end_tracking();
```

Comment:
It will be gone with the next commit anyway, but the original idea was to reset the allocator limit, thus allowing for allocations in the handler, and to try to use a less hacky way to handle it.